blackduck-artifactory-common/src/main/java/com/synopsys/integration/blackduck/artifactory/modules/inspection/externalid/composer/ComposerJsonService.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * blackduck-artifactory-common
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.blackduck.artifactory.modules.inspection.externalid.composer;
+
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.artifactory.repo.RepoPath;
+import org.artifactory.resource.ResourceStreamHandle;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.synopsys.integration.blackduck.artifactory.ArtifactoryPAPIService;
+import com.synopsys.integration.blackduck.artifactory.modules.inspection.externalid.composer.model.ComposerJsonResult;
+import com.synopsys.integration.blackduck.artifactory.modules.inspection.externalid.composer.model.ComposerVersion;
+import com.synopsys.integration.blackduck.artifactory.modules.inspection.externalid.composer.model.FileNamePieces;
+
+public class ComposerJsonService {
+    private final ArtifactoryPAPIService artifactoryPAPIService;
+    private final Gson gson;
+
+    public ComposerJsonService(ArtifactoryPAPIService artifactoryPAPIService, Gson gson) {
+        this.artifactoryPAPIService = artifactoryPAPIService;
+        this.gson = gson;
+    }
+
+    public ComposerJsonResult findJsonFiles(RepoPath repoPath) {
+        FileNamePieces fileNamePieces = extractFileNamePieces(repoPath);
+        String jsonFileName = fileNamePieces.getComponentName().toLowerCase() + ".json";
+        List<RepoPath> foundArtifacts = artifactoryPAPIService.itemsByName(jsonFileName, repoPath.getRepoKey());
+        return new ComposerJsonResult(fileNamePieces, foundArtifacts);
+    }
+
+    public List<ComposerVersion> parseJson(RepoPath repoPath) {
+        try (ResourceStreamHandle resourceStreamHandle = artifactoryPAPIService.getArtifactContent(repoPath)) {
+            InputStream inputStream = resourceStreamHandle.getInputStream();
+            InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
+            JsonElement root = JsonParser.parseReader(inputStreamReader);
+            Set<Map.Entry<String, JsonElement>> rootEntries = root.getAsJsonObject().get("packages").getAsJsonObject().entrySet();
```

Comment:
It doesn't seem like you care about the keys for any of these after "packages"-- could you stream and flatmap over the values you get here, then map into ComposerVersions and collect? I suspect that code would tell a clearer story if you could.